src/interfaces.ts

```diff
@@ -6,50 +6,135 @@ export interface NotificationObject {
   eventCode?: string
 }
 
-interface ContractObject {
-  contractAddress: string
-  contractType: string
+export interface ContractCall {
   methodName: string
-  params: object
+  params: {
+    [key: string]: string
+  }
+  contractAddress?: string
+  contractType?: string
+  contractDecimals: number
+  contractName: string
+  decimalValue: string
 }
 
-export interface TransactionData {
-  asset?: string
-  blockHash?: string
-  blockNumber?: number
-  contractCall?: ContractObject
+export interface CommonTransactionData {
+  system: System
+  network: Network
+  status: Status
+  id?: string
   eventCode: string
-  from?: string
-  gas?: string
-  gasPrice?: string
-  hash?: string
-  txid?: string
-  id: string
-  input?: string
-  monitorId?: string
-  monitorVersion?: string
-  nonce?: number
-  r?: string
-  s?: string
-  status: string
-  to?: string
-  transactionIndex?: number
-  v?: string
-  value?: string | number
+  timeStamp: string
+  serverVersion: string
+  monitorId: string
+  monitorVersion: string
+}
+
+export interface BitcoinTransactionData extends CommonTransactionData {
+  txid: string
+  inputs: InputOutput[]
+  outputs: InputOutput[]
+  fee: string
+  netBalanceChanges: BalanceChange[]
+  rawTransaction: BitcoinRawTransaction
+}
+
+export interface EthereumTransactionData extends CommonTransactionData {
+  hash: string
+  asset: string
+  blockHash: string
+  blockNumber: number
+  contractCall: ContractCall
+  to: string
+  from: string
+  gas: string
+  gasPrice: string
+  gasUsed?: string
+  input: string
+  nonce: number
+  v: string
+  r: string
+  s: string
+  transactionIndex: number
+  value: string
   startTime?: number
+  timePending?: string
   watchedAddress?: string
-  originalHash?: string
+  replaceHash?: string
   counterparty?: string
   direction?: string
-  system?: System
 }
 
+export type TransactionData = BitcoinTransactionData | EthereumTransactionData
+
 export interface TransactionEvent {
   emitterResult: void | boolean | NotificationObject
-  transaction: TransactionData
+  transaction: TransactionData | TransactionEventLog
 }
 
 export type System = 'bitcoin' | 'ethereum'
+export type Network =
+  | 'main'
+  | 'testnet'
+  | 'ropsten'
+  | 'rinkeby'
+  | 'goerli'
+  | 'kovan'
+  | 'xdai'
```

Comment:
Do we need `local` as an option? It's supported on the bn-server

Reply:
Yeah that's a good point, will add it in 👍